api/v1alpha1/managedcluster_types.go

```diff
@@ -61,6 +61,10 @@ type ManagedClusterSpec struct {
 	Template string `json:"template"`
 	// Name reference to the related Credentials object.
 	Credential string `json:"credential,omitempty"`
+	// PropagateCredentials indicates whether credentials should be propagated
+	// for use by CCM (Cloud Controller Manager).
+	// +kubebuilder:default:=true
```

Comment:
```suggestion
	// +kubebuilder:default:=true

	// PropagateCredentials indicates whether credentials should be propagated
	// for use by CCM (Cloud Controller Manager).
```

Sorry for the very nit comment, it is up to you whether to change or not :)